test/unit/events.js

```diff
@@ -201,3 +202,23 @@ test('should have a defaultPrevented property on an event that was prevent from
 
   Events.trigger(el, 'test');
 });
+
+test('should have relatedTarget correctly set on the event', function() {
+  expect(2);
+
+  var el1 = document.createElement('div'),
+      el2 = document.createElement('div'),
+      relatedEl = document.createElement('div');
+
+  Events.on(el1, 'click', function(e){
+    equal(e.relatedTarget, relatedEl, 'relatedTarget is set for all browsers when referring element is set on the event');
+  });
+
+  Events.trigger(el1, TestHelpers.makeMouseEvent('click', relatedEl));
```

Comment:
It looks like makeMouseEvent is still manually setting relatedTarget on the created events, so I might be missing a detail but it seems like essentially the same thing.  I think to get a true real event we would have to use the element's `focus()` API or something like that to trigger a real event where the browser sets the related target. But that gets into integration test land, where I think a simple test to make sure we're not dropping relatedTarget in our code could be good enough.